src/client.rs

```diff
@@ -46,35 +47,46 @@ pub async fn client_main(
                     thread_client.task_request_name = Some(thread_task_name.to_string());
                 }
                 // Invoke the task function.
-                function(&mut thread_client).await;
+                function(&thread_client).await;
```

Comment:
Unrelated whitespace change?

Reply:
This is removing the `mut`. When `&` precedes a variable there's no space. (For the most part, white space is managed by `cargo fmt`)